### PR TITLE
Architecture docs

### PR DESCRIPTION
**Pull request by ylt**

### CODE REVIEW

File: packages/openactive-integration-tests/ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Jest concepts
+
+Jest is multi-process. There's the overall Jest process, which has a bunch of workers which execute the individual test suites (e.g. individual -test.js files). The environment of this main process is different to the worker.
+
+Jest internally uses a fork of Jasmine. This is used as the test runner, and implements standard jest features such as `describe` and `it`.
+
+# Logging/reporting pipeline
+
+Jasmine and Jest as a whole are designed for more binary-like states, a test either passes or fails. With OpenActive the needs a little different to what it's designed for. We generate a lot more debugging level information, and have different levels of failures (i.e. warnings, information, suggestions).
```
Comment (henryaddison) [inline packages/openactive-integration-tests/ARCHITECTURE.md:9]: Small edit.
```suggestion
Jasmine and Jest are designed for binary test states: a test either passes or fails. OpenActive's needs are a little different. More debugging level information is required to help booking system developers act on failures and there are different levels of failures (i.e. warnings, information, suggestions).
```

File: packages/openactive-integration-tests/ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+The process can be summarised as:
+
+- Jasmine reporter: provides state info
+- Logger:  uses state info, used by tests to record info
+- Jest reporter: used to capture test results, and kick off report generation
+- Report generator: generates the reports in both markdown and for cli.
```
Comment (henryaddison) [inline packages/openactive-integration-tests/ARCHITECTURE.md:16]: Link to the relevant modules/classes in the code?

File: packages/openactive-integration-tests/ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+A logger component has been implemented, this deterministically names the filename based on the describe blocks.
+
+This has helper methods for attaching info to the current running test:
+
+- Request:
+    - Saves request (payload, url)
+    - Response (headers, status, response, response time)
+    - Records whether a response was never received (i.e. timedout)
+- Validation:
+    - Validation responses from [https://github.com/openactive/data-model-validator](https://github.com/openactive/data-model-validator), the core library behind [https://validator.openactive.io/](https://validator.openactive.io/)
```
Comment (henryaddison) [inline packages/openactive-integration-tests/ARCHITECTURE.md:33]:
```suggestion
    - Validation responses from [https://github.com/openactive/data-model-validator](https://github.com/openactive/data-model-validator), the same library behind [https://validator.openactive.io/](https://validator.openactive.io/)
```

File: packages/openactive-integration-tests/ARCHITECTURE.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+The process can be summarised as:
+
+- Jasmine reporter: provides state info
+- Logger:  uses state info, used by tests to record info
+- Jest reporter: used to capture test results, and kick off report generation
+- Report generator: generates the reports in both markdown and for cli.
+
+## Jasmine Reporter
+
+This is a state tracker for internal use by the logger, this allows determining within tests the current running test and suite, and also allows hooking the events.
```
Comment (henryaddison) [inline packages/openactive-integration-tests/ARCHITECTURE.md:20]: What does `hooking the events` mean?

File: package.json [hunk 5]
```diff
@@ -9,7 +9,9 @@
     "install": "lerna bootstrap",
     "lerna": "lerna",
     "prettier": "prettier --check \"packages/**/*.js\" \"packages/**/*.json\"",
-    "prettier:write": "prettier --write \"packages/**/*.js\" \"packages/**/*.json\""
+    "prettier:write": "prettier --write \"packages/**/*.js\" \"packages/**/*.json\"",
+    "start": "cd packages/openactive-broker-microservice && npm run start",
+    "test": "cd packages/openactive-integration-tests && npm run test"
```
[the remark this answers is not in the document]
Comment (ylt) [reply package.json:14]: Yeah, wasn't 100% sure how well these work on Windows

**Comment by ylt** [comment]
Logger produces just JSON files. The report generator takes those in and produces the MD/CLI output.

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (henryaddison) [reply package.json:14]: ah OK. Leave it then.

**Pull request merged**